test/spec/helper.clears.js

```diff
@@ -68,6 +71,58 @@ test( "Clearing the same field from multiple elements should remove it everywher
   t.end();
 } );
 
+test( "Clear with a function : neutral and abosorbing ops", function( t ) {
+  var helper = init();
+  var state0 = helper.state;
+
+  helper.clearRefinements( function( value, key, type ) {
+    return false;
+  } );
+
+  t.deepEqual( helper.state.numericRefinements, state0.numericRefinements, "Neutral op : numeric ref should be equal" );
+  t.deepEqual( helper.state.facetsRefinements, state0.facetsRefinements, "Neutral op : conj ref should be equal" );
+  t.deepEqual( helper.state.facetsExcludes, state0.facetsExcludes, "Neutral op : exclude ref should be equal" );
+  t.deepEqual( helper.state.disjunctiveFacetsRefinements, state0.disjunctiveFacetsRefinements, "Neutral op : disj ref should be equal" );
+
+  helper.clearRefinements( function( value, key, type ) {
+    return true;
+  } );
+
+  t.assert( isEmpty( helper.state.numericRefinements ), "remove all numericRefinements" );
+  t.assert( isEmpty( helper.state.facetsRefinements ), "remove all facetsRefinements" );
+  t.assert( isEmpty( helper.state.facetsExcludes ), "remove all facetsExcludes" );
+  t.assert( isEmpty( helper.state.disjunctiveFacetsRefinements ), "remove all disjunctiveFacetsRefinements" );
+
+  t.end();
+} );
+
+test( "Clear with a function : filtering", function( t ) {
+  var helper = init();
```

Comment:
When I read the test I definitely thought the "init()" part was maybe not the best name. It's a fixture let's call it this way?